src/component.js

```diff
@@ -32,11 +32,13 @@ export function Component(props, context) {
 	 */
 	this.props = props;
 
+	this._nextState =
```

Comment:
@KevinDoughty Perhaps a bad push?

Reply:
Actually no, Are those logged type declarations used by some automated system, perhaps for generating docs? Originally I wrote the line as the following:
```JS
	/**
	 * @public
	 * @type {object}
	 */
	this._nextState = this.state = this.state || {}
```
But that implies that `_nextState` is public. I understand how atrocious that looks, it only saves a single byte doing it that way. I could easily spend that byte for nicer looking code, and put it on a separate line:
```JS
	/**
	 * @public
	 * @type {object}
	 */
	this.state = this.state || {};

	this._nextState = this.state;
```